### PR TITLE
[JENKINS-65457] Add description to credentials and ignoreMissing argu…

### PR DESCRIPTION
**Pull request by Onyimatics**
…ments

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- Created `help-credentials.html` file to add `credentials ` argument description. 
- Created `help-ignoreMissing.html` file to add `ignoreMissing ` argument description. 
- [x] Link to relevant issues in GitHub or Jira
- [JENKINS-65457](https://issues.jenkins.io/browse/JENKINS-65457)

 @MarkEWaite @oleg-nenashev @kwhetstone @StackScribe @aHenryJard

### CODE REVIEW

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Select credentials from the list of credentials available to the job.
```
Comment (MarkEWaite) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html:2]: Possible alternate phrasing with two examples included inside the help?
```suggestion
    List of credentials to be used by the sshagent step.
    The Pipeline Syntax Snippet Generator will guide the user to select a single credential from the list of available, like this:
    <code>
    sshagent(credentials: ['my-credential-id']) {
        // some block
    }
    </code>
    The step also accepts a list of credential, like this:
    <code>
    sshagent(credentials: ['my-credential-id', 'my-alternate-credential', 'another-credential']) {
        // some block
    }
    </code>
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Clarification on the text for the credentials field

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html (hunk 1 above)
Comment (Onyimatics) [reply src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html]: Thank you @MarkEWaite

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div>
+    List of credentials to be used by the sshagent step.
+    The Pipeline Syntax Snippet Generator will guide the user to select a single credential from the list of available, like this:
+    <code>
+    sshagent(credentials: ['my-credential-id']) {
+        // some block
+    }
+    </code>
```
Comment (MarkEWaite) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html:8]: I didn't check how the text appears before I proposed my change.  The `<code>` markup does not preserve the line breaks.  We either need explicit `<br/>` or we need the sample to be in a `<pre>` block.
```suggestion
<pre>
sshagent(credentials: ['my-credential-id']) {
    // some block
}
</pre>
```

With that change, now it looks like this:

screencapture-localhost-8080-jenkins-pipeline-syntax-2021-04-24-12_06_28

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    The step also accepts a list of credential, like this:
+    <code>
+    sshagent(credentials: ['my-credential-id', 'my-alternate-credential', 'another-credential']) {
+        // some block
+    }
+    </code>
```
Comment (MarkEWaite) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html:14]:
```suggestion
<pre>
sshagent(credentials: ['my-credential-id', 'my-alternate-credential', 'another-credential']) {
    // some block
}
</pre>
```

Sorry about that mistake!

**Review by MarkEWaite — APPROVED** [review]
Formatting corrections for your consideration.  Sorry that I missed them in my earlier review!

Thanks for the pull request!

**Review by MarkEWaite — APPROVED** [review]

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div>
+    List of credentials to be used by the sshagent step.
+    The Pipeline Syntax Snippet Generator will guide the user to select a single credential from the list of available, like this:
+<pre>
+sshagent(credentials: ['my-credential-id']) {
+    // some block
+}
+</pre>
+    The step also accepts a list of credential, like this:
```
Comment (kwhetstone) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html:9]:
```suggestion
    The step also accepts a list of credentials, like this:
```

**Review by kwhetstone — COMMENTED** [review]
Looks good, I like the examples in the help

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html (hunk 4 above)
Comment (Onyimatics) [reply src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html]: Thanks for the review @Kwhetstone

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<div>
+    When set to true, any missing credentials will be ignored.
+    When set to false, then the build fails if any of the required credentials cannot be resolved.
+    Default is false.
+</div>
```
Comment (oleg-nenashev) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html:5]:
```suggestion
</div>

```

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<div>
+    When set to true, any missing credentials will be ignored.
+    When set to false, then the build fails if any of the required credentials cannot be resolved.
+    Default is false.
```
Comment (oleg-nenashev) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html:4]:
```suggestion
    Defaults to false.
```

**Review by oleg-nenashev — APPROVED** [review]
Just a few minor suggestions

**Review by aHenryJard — APPROVED** [review]
Looks good to me.

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<div>
+    List of credentials to be used by the sshagent step.
```
Comment (MRamonLeon) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-credentials.html:2]:
```suggestion
    List of credentials to be used by the <code>sshagent</code> step.
```

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<div>
+    When set to true, any missing credentials will be ignored.
```
Comment (MRamonLeon) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html:2]:
```suggestion
    When set to <code>true</code>, any missing credentials will be ignored.
```

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+    When set to false, then the build fails if any of the required credentials cannot be resolved.
```
Comment (MRamonLeon) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html:3]:
```suggestion
    When set to <code>false</code>, then the build fails if any of the required credentials cannot be resolved.
```

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+    Defaults to false.
```
Comment (MRamonLeon) [inline src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html:4]:
```suggestion
    Defaults to <code>false</code>.
```

**Review by MRamonLeon — APPROVED** [review]

File: src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html (hunk 10 above)
Comment (Onyimatics) [reply src/main/resources/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep/help-ignoreMissing.html]: Thanks for the review @MRamonLeon

**Pull request merged**